ingestion/src/metadata/orm_profiler/metrics/static/column_count.py

```diff
@@ -15,9 +15,32 @@
 # pylint: disable=duplicate-code
 
 from sqlalchemy import inspect, literal
+from sqlalchemy.ext.compiler import compiles
 from sqlalchemy.orm import DeclarativeMeta
+from sqlalchemy.sql.functions import FunctionElement
 
-from metadata.orm_profiler.metrics.core import StaticMetric, _label
+from metadata.orm_profiler.metrics.core import CACHE, StaticMetric, _label
+from metadata.orm_profiler.orm.registry import Dialects
+
+
+class ColunCountFn(FunctionElement):
+    name = __qualname__
+    inherit_cache = CACHE
+
+
+@compiles(ColunCountFn)
+def _(element, compiler, **kw):
+    return compiler.process(element.clauses, **kw)
+
+
+@compiles(ColunCountFn, Dialects.IbmDbSa)
+@compiles(ColunCountFn, Dialects.Db2)
+def _(element, compiler, **kw):
+    """Returns column count for db2 database and handles casting variables.
+    If casting is not provided for variables, db2 throws error.
+    """
+    proc = compiler.process(element.clauses, **kw)
+    return f"CAST({proc} AS BIGINT)"
```

Comment:
Is this to handle the type not matching when it is None?

Reply:
@TeddyCr it's not for `None`.

The query in db2 does not fail for `NULL` datatypes as we were suspecting earlier.
The query fails when we have variables in the query i.e the `?` thing. If this is not type casted into a DB2 datatype the query fails for DB2
The logic written here and other places is to handle this use case